src/DropdownReference/components/DropdownReferenceContainer.ts

```diff
@@ -0,0 +1,249 @@
+        const context = new mendix.lib.MxContext();
+
+        context.setContext(mxObject.getEntity(), mxObject.getGuid());
+        if (onChangeEvent === "callMicroflow" && onChangeMicroflow) {
+            window.mx.ui.action(onChangeMicroflow, {
+                error: error => window.mx.ui.error(`Error while executing onchange microflow ${onChangeMicroflow}: ${error.message}`), // tslint:disable-line max-line-length
+                origin: mxform,
+                params: {
+                    applyto: "selection",
+                    guids: [ mxObject.getGuid() ]
+                }
+            });
+        } else if (onChangeEvent === "callNanoflow" && onChangeNanoflow.nanoflow) {
+            window.mx.data.callNanoflow({
+                context,
+                error: error => window.mx.ui.error(`Error while executing onchange nanoflow: ${error.message}`),
+                nanoflow: onChangeNanoflow,
+                origin: mxform
+            });
+        }
+    }
+
+    private retrieveOptions(props: ContainerProps) {
+        const entity = this.props.entityPath.split("/")[1];
+        const { sortAttributes, attribute, entityConstraint, source, sortOrder, microflow, mxObject, nanoflow } = props;
+        const attributeReference = `${props.entityPath}${attribute}`;
+        const options: FetchDataOptions = {
+            attributes: [ attributeReference ],
+            constraint: entityConstraint,
+            entity,
+            guid: mxObject.getGuid(),
+            microflow,
+            mxform: this.props.mxform,
+            nanoflow,
+            sortAttributes: !sortAttributes.length
+                ? [ { name: attribute, sort: "asc" } ]
+                : sortAttributes,
+            sortOrder,
+            source
+        };
+
+        fetchData(options)
+            .then(optionObjects => this.setOptions(optionObjects))
+            .catch(errorMessage => window.mx.ui.error(errorMessage.message));
+    }
+
+    private setOptions = (mendixObjects: mendix.lib.MxObject[]) => {
+        const options: ReferenceOption[] = [];
+
+        mendixObjects.forEach(mxObject => {
+            options.push({
+                label: mx.parser.formatAttribute(mxObject, this.props.attribute),
+                value: mxObject.getGuid()
+            });
+        });
+
+        this.setState({ options });
+    }
+
+    private setAsyncOptions = (input: string): Promise<{ options: ReferenceOption[] }> => {
+        if (this.props.mxObject) {
+            this.props.mxObject.set(this.props.searchAttribute, input);
+            if (input.length >= this.props.minimumCharacter) {
+                this.retrieveOptions(this.props);
+
+                return Promise.resolve({ options: this.state.options });
+            } else {
+                this.getSelectedValue(this.props);
+
+                return Promise.resolve({ options: [] });
+            }
+        } else { return Promise.resolve({ options: [] }); }
+    }
```

Comment:
Move the code **return Promise.resolve({ options: [] });** to outside the first else and remove from elses.

private setAsyncOptions = (input: string): Promise<{ options: ReferenceOption[] }> => {
        if (this.props.mxObject) {
            this.props.mxObject.set(this.props.searchAttribute, input);
            if (input.length >= this.props.minimumCharacter) {
                this.retrieveOptions(this.props);
                 return Promise.resolve({ options: this.state.options });
            } else {
                this.getSelectedValue(this.props);
            }
        }
       return Promise.resolve({ options: [] });
    }

Reply:
Thanx. Fixed.